torch/csrc/jit/fusion_compiler.cpp

```diff
@@ -624,27 +641,179 @@ void compressContiguous(
 
 } // anonymous namespace
 
+// XXX: assumes graph inputs are 32-bit indexable
+// assumes v is an input to the graph.
+static size_t toInputIndex(Value * v, Graph & subgraph) {
+  auto start = subgraph.inputs().begin();
+  auto it = std::find(start, subgraph.inputs().end(), v);
+  JIT_ASSERT(it != subgraph.inputs().end());
+  return it - start;
+}
+
+// If there are fused Chunk nodes in the graph, returns a new Graph replacing
```

Comment:
If you look at how this is handled for concat, it takes a more lightweight approach. By leaving the Concat operators in and just skipping them in code generation we can still generate the correct kernel, and make sure the input list has the 'flattened' inputs added, without having to copy/do surgery on the fusion graph. It is possible that something similar might work here (see later comment about how chunk and concat are basically the same exact operator).